app/src/main/java/fi/thl/koronahaavi/exposure/ManualCheckAllowedLiveData.kt

```diff
@@ -0,0 +1,25 @@
+package fi.thl.koronahaavi.exposure
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MediatorLiveData
+import fi.thl.koronahaavi.device.SystemState
+
+class ManualCheckAllowedLiveData(
+    private val systemState: LiveData<SystemState?>,
+    private val exposureState: LiveData<ExposureState>,
+    private val checkInProgress: LiveData<Boolean>
+) : MediatorLiveData<Boolean>() {
+
+    init {
+        addSource(systemState) { updateManualCheckAllowed() }
+        addSource(exposureState) { updateManualCheckAllowed() }
+        addSource(checkInProgress) { updateManualCheckAllowed() }
+    }
+
+    private fun updateManualCheckAllowed() {
+        // sync showManualCheck value with checkInProgress so that button and progress indicator disappear at the same time
+        value = (checkInProgress.value == true) ||
```

Comment:
Is this condition correct - shouldn't manual check be allowed only if checkInProgress is 'false', or is somehow related to the mentioned 'syncing'?

Reply:
The class name is confusing, it should actually be ManualCheckShownLiveData, because when check is in progress, the button is shown, but it's disabled... this got a bit complicated and we could make it simpler by building a custom button control that integrates the progress indicator but thats a bit of work

Reply:
renamed the class